### PR TITLE
uniparc light model and builder

### PR DESCRIPTION
**Pull request by ahmadshadab**

### CODE REVIEW

File: controlled-vocabulary/src/test/java/org/uniprot/cv/xdb/UniProtKBCrossReferenceDisplayOrderTest.java [hunk 1]
```diff
@@ -4,9 +4,11 @@
 
 import java.util.List;
 
+import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
 import org.uniprot.core.cv.xdb.UniProtDatabaseDetail;
 
+@Disabled
```
Comment (LeonardoGonzales) [inline controlled-vocabulary/src/test/java/org/uniprot/cv/xdb/UniProtKBCrossReferenceDisplayOrderTest.java:11]: We restored it. https://www.ebi.ac.uk/uniprot/prod/generator/dr_ord it should be working now.

File: core-domain/src/main/java/org/uniprot/core/uniparc/UniParcEntryLight.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+import org.uniprot.core.Sequence;
+import org.uniprot.core.util.Pair;
+
+import java.io.Serializable;
+import java.time.LocalDate;
+import java.util.List;
+
+public interface UniParcEntryLight extends Serializable {
+    UniParcId getUniParcId();
```
Comment (LeonardoGonzales) [inline core-domain/src/main/java/org/uniprot/core/uniparc/UniParcEntryLight.java:11]: Can we use String for this as well?

File: core-domain/src/main/java/org/uniprot/core/uniparc/UniParcEntryLight.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    LocalDate getMostRecentCrossRefUpdated();
+
+    // lazily loaded fields
+    List<Pair<Integer, String>> getOrganisms();// organism id and name pair
+
+    List<String> getProteinNames();
+
+    List<String> getGeneNames();
+
+    List<String> getProteomeIds();
```
Comment (LeonardoGonzales) [inline core-domain/src/main/java/org/uniprot/core/uniparc/UniParcEntryLight.java:35]: I think proteomes will need to be a pair as well, id and component, and I think that sometimes we do not have the component.  (https://www.uniprot.org/uniparc?fields=upi%2Corganism%2Clength%2Caccession%2Cfirst_seen%2Clast_seen%2Cproteome&query=P05067&view=table)

Comment (LeonardoGonzales) [reply core-domain/src/main/java/org/uniprot/core/uniparc/UniParcEntryLight.java:35]: Do you think we should rename it just to getProteomes() ?

File: core-domain/src/main/java/org/uniprot/core/uniparc/impl/UniParcEntryLightImpl.java [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+    }
+    UniParcEntryLightImpl(UniParcId uniParcId, List<String> uniParcCrossReferences,
+                                 List<Pair<Integer, String>> commonTaxons, List<String> uniProtKBAccessions,
+                                 String uniProtExclusionReason, Sequence sequence, List<SequenceFeature> sequenceFeatures,
+                                 LocalDate oldestCrossRefCreated, LocalDate mostRecentCrossRefUpdated,
+                                 List<Pair<Integer, String>> organisms, List<String> proteinNames, List<String> geneNames,
+                                 List<String> proteomeIds) {
+        this.uniParcId = uniParcId;
+        this.uniParcCrossReferences = uniParcCrossReferences;
+        this.commonTaxons = commonTaxons;
```
Comment (LeonardoGonzales) [inline core-domain/src/main/java/org/uniprot/core/uniparc/impl/UniParcEntryLightImpl.java:58]: I think we could use Utils.unmodifiableList for list items

File: core-domain/src/main/java/org/uniprot/core/uniparc/UniParcEntryLight.java (hunk 2 above)
Comment (supun-ebi) [inline core-domain/src/main/java/org/uniprot/core/uniparc/UniParcEntryLight.java:11]: I also wonder whether separating the value and validation logic is possible.

File: core-domain/src/main/java/org/uniprot/core/uniparc/UniParcEntryLight.java (hunk 3 above)
Comment (ahmadshadab) [reply core-domain/src/main/java/org/uniprot/core/uniparc/UniParcEntryLight.java:35]: Based on our discussion, we agree to return proteome id for now. If asked, we will add it.

File: core-domain/src/main/java/org/uniprot/core/uniparc/UniParcEntryLight.java (hunk 2 above)
Comment (ahmadshadab) [reply core-domain/src/main/java/org/uniprot/core/uniparc/UniParcEntryLight.java:11]: Which validation logic are you talking about?

Comment (ahmadshadab) [reply core-domain/src/main/java/org/uniprot/core/uniparc/UniParcEntryLight.java:11]: Sure, made it to String.

**Comment by LeonardoGonzales** [comment]
I think we can close this PR, right?

**Pull request closed without merging**